python/arcticdb/adapters/mongo_library_adapter.py

```diff
@@ -0,0 +1,63 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+from arcticdb.options import LibraryOptions
+from arcticc.pb2.storage_pb2 import EnvironmentConfigsMap, LibraryConfig
+from arcticdb.version_store.helper import add_mongo_library_to_env
+from arcticdb.config import _DEFAULT_ENV
+from arcticdb.version_store._store import NativeVersionStore
+from arcticdb.adapters.arctic_library_adapter import ArcticLibraryAdapter, set_library_options
+from arcticdb_ext.storage import Library
+from arcticdb.encoding_version import EncodingVersion
+from arcticdb.exceptions import UserInputException
+from collections import namedtuple
+from dataclasses import dataclass, fields
+import pymongo
+
+PARSED_QUERY = namedtuple("PARSED_QUERY", ["region"])
+
+
+class MongoLibraryAdapter(ArcticLibraryAdapter):
+    @staticmethod
+    def supports_uri(uri: str) -> bool:
+        return uri.startswith("mongodb://")
+
+    def __init__(self, uri: str, *args, **kwargs):
+        try:
+            parameters = pymongo.uri_parser.parse_uri(uri)
+            self._endpoint = f"{parameters['nodelist'][0][0]}:{parameters['nodelist'][0][1]}"
+        except Exception as e:
```

Comment:
At first I thought `Exception` was too broad, but `pymongo` is rubbish at this so not much we can do:

```
>>> uri = "blah not a uri"
>>> pymongo.uri_parser.parse_uri(uri)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/var/aseaton/tst1/lib/python3.8/site-packages/pymongo/uri_parser.py", line 433, in parse_uri
    raise InvalidURI("Invalid URI scheme: URI must "
pymongo.errors.InvalidURI: Invalid URI scheme: URI must begin with 'mongodb://' or 'mongodb+srv://'
>>> pymongo.uri_parser.parse_uri("mongodb://notgreat")
{'nodelist': [('notgreat', 27017)], 'username': None, 'password': None, 'database': None, 'collection': None, 'options': {}, 'fqdn': None}
>>> pymongo.uri_parser.parse_uri("mongodb://b:a:d:u:r:i")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/var/aseaton/tst1/lib/python3.8/site-packages/pymongo/uri_parser.py", line 514, in parse_uri
    nodes = split_hosts(hosts, default_port=default_port)
  File "/var/aseaton/tst1/lib/python3.8/site-packages/pymongo/uri_parser.py", line 364, in split_hosts
    nodes.append(parse_host(entity, port))
  File "/var/aseaton/tst1/lib/python3.8/site-packages/pymongo/uri_parser.py", line 111, in parse_host
    raise ValueError("Reserved characters such as ':' must be "
ValueError: Reserved characters such as ':' must be escaped according RFC 2396. An IPv6 address literal must be enclosed in '[' and ']' according to RFC 2732.
```